### DISCUSSION TITLE
q-scroll-area and q-list are used together, but there is a bug

### DISCUSSION THREAD

**Discussion by 332lfgb6** [quasarframework/quasar#15090] [General - Components / Directives / etc] [answered]
https://codepen.io/332lfgb6/pen/WNyBNBQ

Instead of a horizontal scroll bar, demo2 can be omitted like demo1, line property, seems to have a problem

![image](https://user-images.githubusercontent.com/51245335/206912097-98ec44a1-9732-4278-914b-57841b9691fd.png)

**Comment by pdanpdan** [comment]
I suppose what you want is to have a vertical-only QScrollArea?
https://codepen.io/pdanpdan/pen/dyKEYmm
You need to set a `content-style="width: 100%"`

**Reply by 332lfgb6** [reply]
> I suppose what you want is to have a vertical-only QScrollArea? https://codepen.io/pdanpdan/pen/dyKEYmm You need to set a `content-style="width: 100%"`

nonono!
![image](https://user-images.githubusercontent.com/51245335/206925943-76d6840b-86b7-4cb8-b599-9046dcaf3e56.png)
If you look at my code, I set line=1, but it doesn't work

**Comment by pdanpdan** [comment]
What do you mean by line 1 does not work?
You only have one line

**Reply by 332lfgb6** [reply]
> What do you mean by line 1 does not work? You only have one line

The code for my project looks like this (I'd love to show you in codepen, but he can't)

```vue
<q-layout>
    <q-page-container>
      <q-scroll-area class="scroll-area">
        <q-page>
          <div class="row">
            <div class="col-12">
              <q-card>
                <q-list>
                  <q-item v-ripple clickable>
                    <q-item-section>
                      <q-item-label lines="1">I know that you are alone, alone tonight, but in my heart</q-item-label>
                    </q-item-section>
                  </q-item>
                </q-list>
              </q-card>
            </div>
          </div>
        </q-page>
      </q-scroll-area>
    </q-page-container>
  </q-layout>

<style lang="scss" scoped>
  .scroll-area {
    height: 300px;
    width: 300px;
  }
</style>
```

#### Preview

![image](https://user-images.githubusercontent.com/51245335/206927136-5dec45d4-ce87-4f30-bd7a-d652863869ba.png)


When I add content-type="width: 100%", line=1 still does not take effect

```vue
<q-layout>
    <q-page-container>
      <q-scroll-area class="scroll-area" content-style="width: 100%">
        <q-page>
          <div class="row">
            <div class="col-12">
              <q-card>
                <q-list>
                  <q-item v-ripple clickable>
                    <q-item-section>
                      <q-item-label lines="1">I know that you are alone, alone tonight, but in my heart</q-item-label>
                    </q-item-section>
                  </q-item>
                </q-list>
              </q-card>
            </div>
          </div>
        </q-page>
      </q-scroll-area>
    </q-page-container>
  </q-layout>

<style lang="scss" scoped>
  .scroll-area {
    height: 300px;
    width: 300px;
  }
</style>
```

#### Preview
Same thing as above
![image](https://user-images.githubusercontent.com/51245335/206927136-5dec45d4-ce87-4f30-bd7a-d652863869ba.png)

## lines="1" did't work, How can I change the code?

**Comment by pdanpdan** [comment]
You have a QPage in QScrollArea - first thing would be the change the order
Check in inspector your elements - see which one extends to fit the content.
Lines 1 has nothing to do with it - it just sets ellipsis overflow in the element, but if it has no maximum width defined of course it does nothing

**Reply by 332lfgb6** [reply]
> You have a QPage in QScrollArea - first thing would be the change the order Check in inspector your elements - see which one extends to fit the content. Lines 1 has nothing to do with it - it just sets ellipsis overflow in the element, but if it has no maximum width defined of course it does nothing

https://jsfiddle.net/6zx7mbfy/24/

I tried many times, But I still don't know why lines=1 doesn't work, Can you have a look at my new code?

**Reply by 332lfgb6** [reply]
> You have a QPage in QScrollArea - first thing would be the change the order Check in inspector your elements - see which one extends to fit the content. Lines 1 has nothing to do with it - it just sets ellipsis overflow in the element, but if it has no maximum width defined of course it does nothing

I want an ellipsis, not a horizontal scroll bar.
![image](https://user-images.githubusercontent.com/51245335/207073527-7e398dc3-74df-4689-abfb-23cfaeaee441.png)
My English is so bad, I don't know if you can understand me😄

**Comment by pdanpdan** [comment, marked as answer]
In your case you should propagate the QScrollArea with to the content
https://jsfiddle.net/e2jz1huv/

**Reply by 332lfgb6** [reply]
> In your case you should propagate the QScrollArea with to the content https://jsfiddle.net/e2jz1huv/

Tks, bro
